Vocable 361 -  Updates to Categories Screen

stevefosterwta opened the pull request:
https://github.com/willowtreeapps/vocable-ios/issues/361

From the Github issue:
- The title "Categories" should change to "Categories and Phrases" on the Settings main screen.
- The hide category toggle should be brought to the Category main screen beside the up and down arrows. When a category is toggled off, it should be moved to the end of the list and the ">" button and up and down arrows should be inaccessible, at 50% opacity.
- For default categories (non-custom categories), the ">" icon that accesses the detail screen should be inaccessible, at 50% opacity, as these cannot be edited.

stevefosterwta committed:
Added the hide category toggle to the Category edit screen

stevefosterwta committed:
Added the hide category toggle to the Category edit screen

stevefosterwta committed:
Set category details access

stevefosterwta committed:
Set category details access

SashankPatel committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/vocable-361'

SashankPatel commented:
Test Notes 8/20:
- Button label has been updated in the Settings screen - Pass
- Hide Category button has been brought out to the Category screen. - Pass
- > button for Default categories is disabled all the time. - Pass
- When hide toggle is pressed (to hide), the category moves to the end of the list. - Pass
- When hide toggle is pressed (to show), the category moves to the order they were in prior to hiding.

Pending task for test: Fix Failing Automation tests

SashankPatel committed:
Fixed failing UI test due to show hide toggle.

Merged in changes from develop and fixed UI tests that were failing due to the show/hide toggle rework.

Clstroud commented on VocableUITests/Tests/MainScreenTests.swift:
```diff
@@ -39,15 +39,14 @@ class MainScreenTests: BaseTest {
+        settingsScreen.toggleHideShowCategory(category: generalCategoryText, toggle: "Show")
 
-        settingsScreen.leaveCategoryDetailButton.tap()
+        //settingsScreen.leaveCategoryDetailButton.tap()
```
```suggestion
```

Clstroud commented on VocableUITests/Screens/SettingsScreen.swift:
```diff
@@ -25,4 +25,25 @@ class SettingsScreen {
+    } else {
+            XCUIApplication().buttons["bottomPagination.right_chevron"].tap()
+            XCUIApplication().collectionViews.cells.otherElements.containing(.staticText, identifier: category).buttons[toggleLabel].tap()
+
```
```suggestion
```

Clstroud commented on VocableUITests/Screens/SettingsScreen.swift:
```diff
@@ -25,4 +25,25 @@ class SettingsScreen {
+        
+        if XCUIApplication().collectionViews.cells.otherElements.containing(.staticText, identifier: category).element.exists {
+            XCUIApplication().collectionViews.cells.otherElements.containing(.staticText, identifier: category).buttons[toggleLabel].tap()
+    } else {
```
```suggestion
        } else {
```
